src/client/components/logo/logo.tsx

```diff
@@ -0,0 +1,28 @@
+import type { Location, History } from 'history';
+import PropTypes from 'prop-types';
+import React, { Validator } from 'react';
+import { match, RouteComponentProps, withRouter } from 'react-router-dom';
+
+class Logo extends React.Component<RouteComponentProps> {
```

Comment:
I think that's a good idea in general, as function components seem to be favored over class components in modern react.

But I'm not sure if it needs to be done at the same time as the typescript migration. I left this particular component as it is (while converting others to function components), so that there are some examples for how to do both ways in TypeScript.

Reply:
Ohhh...  stupid typo from on my side:  Forgot the (relatively important) *not*  in the comment above. Yes, we should not do it at the same time as the TypeScript migration...  ;)